service/history/workflowExecutionContext.go

```diff
@@ -430,3 +430,29 @@ func (c *workflowExecutionContext) updateWorkflowExecutionWithRetry(
 func (c *workflowExecutionContext) clear() {
 	c.msBuilder = nil
 }
+
+func (c *workflowExecutionContext) scheduleNewDecision(transferTasks []persistence.Task,
+	timerTasks []persistence.Task) ([]persistence.Task, []persistence.Task, error) {
+	msBuilder, err := c.loadWorkflowExecution()
+	if err != nil {
+		return transferTasks, timerTasks, err
```

Comment:
nil, nil, err

Reply:
i see the point that returning the input transfer task and timer task to caller, for convenience, however, this too some degree is surprising, since if the return err is not nil, other returned parameters should be considered invalid

Reply:
Although returning nil for transfer and timer tasks makes sense but it will make the code even uglier.  The issue I need to return them in the first place is to accept a slice and append more tasks to it and return the new slice with appended values.  The pattern is to pass in a slice and then store the return arguments to the same slice.  Setting it to nil on error will result in corrupting the original slice unless I store the result in a temporary variable and them store back to original variable if the error is not nil.  I prefer to leave this as is unless you have other objections.

Reply:
please add some comments then, since this is not the common behavior usually seen in the code.